networkx/linalg/algebraicconnectivity.py

```diff
@@ -95,19 +83,7 @@ class _CholeskySolver:
     """
 
     def __init__(self, A):
-        if not self._cholesky:
-            raise nx.NetworkXError("Cholesky solver unavailable.")
-        self._chol = self._cholesky(A)
-
-    def solve(self, B, tol=None):
-        return self._chol(B)
-
-    try:
-        from scikits.sparse.cholmod import cholesky
-
-        _cholesky = cholesky
-    except ImportError:
-        _cholesky = None
+        raise nx.NetworkXError("Cholesky solver deprecated.  Use LU solver instead.")
```

Comment:
Given the dependency on `scikits.sparse` for this particular code path, I'm +1 on taking it out now instead of the standard 1-release deprecation cycle.

Reply:
That works for me.  But I will wait to see what @dschult and others think.

Reply:
It's unlikely to inconvenience people to remove it. So, I'm OK to remove it now.
I think removal may require removing the class and adjusting docs in many places within this module. If you prefer to do that in another PR thats fine too.